log/sequencer.go

```diff
@@ -167,7 +169,19 @@ func (s Sequencer) initMerkleTreeFromStorage(ctx context.Context, root *types.Lo
 		hashes[i] = node.Hash
 	}
 
-	return compact.NewTreeWithState(s.hasher, root.TreeSize, hashes, root.RootHash)
+	rng, err := fact.NewRange(0, root.TreeSize, hashes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create compact.Range: %v", err)
+	}
+	hash, err := rng.GetRootHash(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to restore the root hash: %v", err)
```

Comment:
Not sure about "restore" here. Can maybe pass the error through directly.

Reply:
Changed to "compute".